### ISSUE TITLE
Status of Psi Probe 3.0.0 Release

### ISSUE THREAD

**Issue by hazendaz** [psi-probe/psi-probe#657]
This question came up separately and I wanted to address it.  Currently there really isn't any official stance on the milestone releases I pushed out.  For the two I pushed out, it was more for user community to have easier access to begin testing immediately on work we are doing.  I expect many more to come as we solidify things and certainly correct or add more functionality.

As to when 3.0.0 will be released.  My thought is that we wait until Tomcat releases their version 9 officially since we are otherwise supporting it.  If Tomcat 8 milestones are any indication and I'm sure 7 was the same, Tomcat team did change things along the way that affected the api usage.  So I'd hate to release 3.0.0 only to keep patching as tomcat affects development along the tomcat 9 lines.

One additonal reason to wait is that tomcat 6 support was dropped here.  The longer we wait the closer we get to the official dates Tomcat annouced for tomcat 6 EOL which is the end of this year last I checked.

What I don't know for certain is when tomcat 9 will be official.  That has more to do with when the servlet 4.0 spec is completely finalized.  I can only assume it's still highly work in progress at least at this time.  If anyone has any further information on the subject or feels we need to move a little faster towards release, please let it be known.  My best guess is officially it will be around fall 2016 where we cut the trigger.  If tomcat 9 isn't ready, we may cut the triggers as release candidates to the various release channels.

**Comment by MALfunction84** [comment]
I think nightlies or better yet continuous builds (#499) would be best for users who want to test stuff on the bleeding edge. Is there a way to have a placeholder pre-release that basically is just a link to the latest snapshot build? That'd be less work for everyone, I think.

For issue organization, and especially being able to reproduce issues, it's nice to have discrete version numbers for each release. There are a number of issue labels I've been using to track what version an issue shows up in. Useful for to discover "known issues" in a specific release.

As for scoping Tomcat 9, we can always cut a 3.1 if we beat them to release. :)

**Comment by hazendaz** [comment]
@MALfunction84 I can work on getting travis CI setup later this week to auto deploy to sonatype.  From there, presumably I get the naming right or close to right, users can download off our main readme.  I can look a little further to see if that can be auto released to releases as probe.war (the easiest to use name!).  I assume that is quite possible but haven't yet done that type of release myself.  Any assistance there would be appreciated.  Maven is more my area and basically what I have setup on the mybatis projects should be a drop in place here.  I've been meaning to get that all setup now that the links to maven central are in place.  As soon as a few other things clear to master, I'll start that work on the maven side.

**Comment by diogosantana** [comment]
Can you guys release M4 so I can take advantage of #659 and #644?

**Comment by hazendaz** [comment]
I'll push one tonight


Sent from Outlook Mobile






On Thu, Mar 3, 2016 at 8:17 AM -0800, "diogosantana" <notifications@github.com> wrote:





Can you guys release M4 so I can take advantage of #659 and #644?

---

**Comment by hazendaz** [comment]
Unless anyone else objects, once tomcat 8.5.x makes it out of beta which is going to be significantly sooner than tomcat 9 by the looks of it, I think it will be time to cut official 3.0.0 release.  Please chime in for any thoughts on this.  If we go this route, I'm not sure we will be adding any more features unless those come soon...

**Comment by MALfunction84** [comment]
"Object" is a strong word, but I think we should at least resolve [issues with milestone 3.0.0](/psi-probe/psi-probe/milestones/3.0.0) before releasing 3.0.0. Part of the reason it's 3.0.x and not 2.5.x is because we were planning to do some big changes under the hood.

**Comment by MALfunction84** [comment]
If people want the latest and greatest, that's what the snapshots are for. What's your hurry?

**Comment by hazendaz** [comment]
Tomcat 9 is effectively being released as 8.5.  That's all.  Me personally....I've been running variations of 3.0.0 in production environments since before november.


Sent from Outlook Mobile






On Sun, Mar 20, 2016 at 10:49 PM -0700, "Mark Lewis" <notifications@github.com> wrote:





If people want the latest and greatest, that's what the snapshots are for. What's your hurry?

---

**Comment by hazendaz** [comment]
Possibly looking at a milestone 4 release in next couple of weeks.  Enough people are checking things now and there have been enough rather significant vulnerabilities with some third party libraries that the time is right to make another push.

**Comment by MALfunction84** [comment]
SGTM

**Comment by hazendaz** [comment]
couple of weeks becomes 6 months.  note to self 'don't make such statements'.  blaming spring on holdup forcing our hand at annotations :(

**Comment by hazendaz** [comment]
3.0.0.RC1 was released with a number of 405 issues still present.  Those have since been fixed and a lot of spring xml is now java config based.  I'm waiting on one user input to outstanding failure before I cut RC2.  Then I plan to introduce experimental Oshi project and will document as such for RC3 and likely the final release then.  I'm still trying to look at additional issues but likely to call this complete soon and simply roll this out then start a new process of monthly releases.

**Comment by hazendaz** [comment]
Closing as 3.0.0 released a while back and 3.1.0 in the works with spring5 / java 8.
